device/core/src/internal_client.ts

```diff
@@ -337,6 +337,42 @@ export abstract class InternalClient extends EventEmitter {
     }, abandonCallback);
   }
 
+  updateClientProperties(propertyCollection: ClientPropertyCollection): Promise<void>;
+  updateClientProperties(propertyCollection: ClientPropertyCollection, done: (err: Error) => void): void;
+  updateClientProperties(propertyCollection: ClientPropertyCollection, done?: (err: Error) => void): Promise<void> | void {
+    return callbackToPromise((_callback) => {
+      this.getTwin((err, twin) => {
+        if (err) {
+          _callback(err);
+          return;
+        }
+        twin.properties.reported.update(propertyCollection.backingObject, _callback);
+      });
+    }, done);
+  }
+
+  onWritablePropertyUpdateRequest(callback: (properties: ClientPropertyCollection) => void): void {
+    this.getTwin((err, twin) => {
+      if (err) {
+        this.emit('error', err);
+        return;
+      }
+      twin.on('properties.desired', (patch) => {
```

Comment:
The way this is implemented, the user can only have a listener for entire desired property patches. In contrast, the legacy twin API lets the user listen for specific properties.

I think this is the approach we want, but would like to hear thoughts on this.

Reply:
This way is better. I think the way we did it for the legacy API was a mistake in this regard.  No other SDK allows for per-property events.